src/ui/public/kuery/lib/kuery.peg

```diff
@@ -0,0 +1,55 @@
+Query
+  = head:Clause tail:(whitespace Clause)* {
+      var result = [head], i;
```

Comment:
Yeah that's a great idea, I had played around with a similar idea in an earlier revision. I'll take this feedback onboard when I flesh out the language more in my next PR.